iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/client/IoTDBDataNodeAsyncClientManager.java

```diff
@@ -233,7 +234,7 @@ public void onError(Exception e) {
           callback);
       waitHandshakeFinished(isHandshakeFinished);
     }
-
+    client.setShouldReturnSelf(true);
     if (exception.get() != null) {
       throw new PipeConnectionException("Failed to handshake.", exception.get());
```

Comment:
If there are exceptions, maybe we should return the client, or else it's leaked and cannot be borrowed again...